File: FitnessTracker/app/src/main/java/com/example/fitnesstracker/MainActivity.java
package com.example.fitnesstracker;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

    }

    public void navigateToSensorActivity(View v) {
        Intent mIntent = new Intent(this, SensorActivity.class);
        startActivity(mIntent);
    }

    public void navigateToHistoryActivity(View v) {
        Intent mIntent = new Intent(this, HistoryActivity.class);
        startActivity(mIntent);
    }

    public void navigateToSubscriptionActivity(View v) {
        Intent mIntent = new Intent(this, SubscriptionActivity.class);
        startActivity(mIntent);

    }
}

File: dataBinding2/app/src/main/java/com/example/databinding/MainActivity.kt
package com.example.databinding

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.example.databinding.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private lateinit var viewModel: MainActivityViewModel
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        viewModel = ViewModelProvider(this).get(MainActivityViewModel::class.java)
        binding.lifecycleOwner = this
        binding.myViewModel = viewModel
    }
}
